src/Fantomas.Core.Tests/MultilineBlockBracketsOnSameColumnRecordTests.fs

```diff
@@ -1270,3 +1270,76 @@ let ``comment after equals in anonymous record field`` () =
         B
 |}
 """
+
+[<Test>]
+let ``inline anonymous record type declaration`` () =
+    formatSourceString
+        false
+        """
+type Foo =
+    {
+        Bar : {| X : string; Y : int; A : string; B : string |}
+        Baz : int
+        Blip : string
+    }
+"""
+        config
+    |> prepend newline
+
+    |> should
+        equal
+        """
+type Foo =
+    {
+        Bar : {|
```

Comment:
Hmm, I'm not quite sure about this part.
Going to the next line indented looks more consistent with a record instance.
[Sample](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXMAxWsYAvADoBOJAdhBMOVTY2AEICGZxYwAPvdUwIEANTiUgBnLGQCWlAOZi%2Bg5TQCanAAxKVygIKiJU2QsjadA5gckz5iiv3NhuAXzNM2AL04BGN42bo0gAOVka2pvaMzpAANOBQ4thYxuKoYADa8bRZNJAAJFgAnkHIaJCyWLE5EPkAbqzoAK6lGdU0GnHKkACSlEiUWADK0h7InYIALG0AujnO49kC%2BUUlaeUDVUt59U0tmcreC0yQALKsAB4AMrKIl4jyWAAWm4LeAEwaM3NHoFsrLZB7vAAPIAM2ulEQgyKmBejDqDWaaX2gjeR3hAFE%2BmCIWM2jV0KDIF8oj82stigCAEYEdBw3LbRF7fFgADM6NyvUSZCwuFkDQAcogAO6BSH0xhSZokmjzHK-Y4FSlrGmwOkchG7ZEsiYcmqDIKsaCIZiIUGwMiIAAK7FYAFtsIgyBKaFLEDKIHLGAr4UrVmVVeryYytWgUQIAKx6qAGo0ms0W26wYVO6CsRK9WqwNPJWCUF0QN0esBemg%2Bhn-FW0l2apFhlkANmjkFjxtN5stAFUgiUyGmM5QsznpHmC2BQQ1EsXS4tFZWA9WNSG661lAB2Zut%2BMdxAAYXQ6fEu7z1ka0CwFrHE-QU4Es1J8uD86ggZry%2BZygAHJvDW2E5aTkQO0qSdK9J3dO9vkfP5lQXNU3x2FdwyYABOH843bRNj3QUdo2vW8mHvWUyRg-0X0XYNEI-V4OhZFtf0QXRQSwJ1jztO1WDHItIIfb0n1g8j4KXKjtQOQ46K3TDLUGIDpDgHD8zw8DpxIucBMgV9hKZUTXjRCSGKYliyBku05LVXCWW4wioL40jqQorYRPrA52X0uNdDIWBGj6AARRBAlMoyuLIaUeOI6C1LI9ZKi00NV1eXU6LOc5ulBAAVR57kGR4LSwAB1aR4CeMdPjCz1VN9Z9ooQ7TnNeKMkouVKMvuDEbyhHKeQKornmjBtSus3iy34qKKhquLkMYbwm0alLKFBaRzmBXtWAvMgMXOIJLXEcQR0UllPwGxgiPKiLKvUsbYqQllvA3WaACVEDgMh4G64rowmI6aBOksKorC6Niu6iBG8b8Hqei14AFRpgKdMFuhYu1UmjbwVLO-6ortRp0GSMVEHwMgOKwIy0uVIGdJBtC6Me574BObHcZuAmiaC5toEeW1zydAB9YVCuKtHbMigFLso2r4oED5m2SjyyFYQpgTIa5JDe3qDq%2BiAfpnLgRpFwGxYmlk3nEroZbIOWFaV6RJGh2GyHhxHkZuwXhrstYsZx6Q8eZ1aSbJg3ruUN49K6WX5cV5WsHpz3vYtFnQLZjm5a5shef554Xdnc7Rv1xzxcmmg3lc02LgANSZZhZHgYxVbHQ7M51t2ylFvPDaDxKS-OXBvPPPbK76Gv07HT6G-LGo-T1mKA%2BBpg3gazufNgLAAHFsA2rbEB2vba%2Bjeuyt%2B9Hx6qzTp4p2eZtNhmvZuAJswAa2YZO7%2BwcRgUoQZ7T3NUYf25QrOOmyrthZViEqfOqks7pdCFKKG%2B2AUz3FJiUPyC1KDSFzJQXQfRAJ2ydn-EKEFBrhSFtneyoDW6B1RGDUOgQ5CUG7pQXueZhi0NWo0S0aVYC9H6FgVae0wLtVHrrEBQZyEz0YG8Km1CjKUF4bUJM8hsEgTIMg2QaC9q4MEPhAhAChpZwxqQkRionISyYKyWil9Pa4kuPaKk8BWD7lgLteQ0C8b8IIjoohQCSHCPGhQgQrITaCEgOvJ00gHQDAaAAaUQIgIIXD7hYF6I-VgDC1bKC0YIpuglDG%2BmMQXCArIQ5BICCku%2BuJxAeS8n0IUkhEB0yvnjde21donmCqFQhp1iH6J8eTcBpji7FPYFJRAPlrbQBkKZGRLF4CdlQXmPy0ADxy3QW47R31AF6KPupE%2Boiz6MFZB3IJISJkJIaNCTyjRTxBGhIUWESkBH721mPKAE81gtyMfnFkrJ55BOibE5KtslFghKZQMpNwNEgw0BrMAWs-pbKijsj5bdBCsgvkE85cko4IDxOk5Sjz4jTHGOUcQuBdppAyc4IAA)

Reply:
Yeah, full honesty that's what I was going for, but I couldn't get it to work before I left on my trip. Everything I tried resulted in broken code 😬. I'll have to take another look when I get home in a few days.

Reply:
You probably want to use `autoIndentAndNlnIfExpressionExceedsPageWidth` or `sepSpaceOrIndentAndNlnIfExpressionExceedsPageWidth` to cover that.

Reply:
Okay so I was able to get it so it's not broken code anymore, and I can get it to print correctly for _either_ `anonymous type alias` _or_ `inline anonymous record type declaration`, but I'm having trouble figuring out how to make it handle the difference between those cases.

If I indent and newline (e.g. with `autoIndentAndNlnIfExpressionExceedsPageWidth` or `indentSepNlnUnindent`), it adds an unnecessary extra newline to the `anonymous type alias` test case. 

Should I be checking e.g. if it's in the rhs of a type abbreviation? Or should I adjust the source for the type abbreviation itself? Or is there another better way to handle that complexity? Thanks for your help!

Reply:
So changing:

![image](https://user-images.githubusercontent.com/2621499/185669526-818da3e9-65ee-4302-b790-16c8e1accacd.png)

fixes the failing test. One impacted test of this change is:

![image](https://user-images.githubusercontent.com/2621499/185669616-024ece21-fddd-4b2c-9dc0-152a289f7077.png)

This looks correct to me but that makes this a bit more difficult to merge into the master branch.

As for 
```fsharp
let f
    (x : {|
             x : int
             y : AReallyLongTypeThatIsMuchLongerThan40Characters
         |})
```

you need to do something similar in `genPat` for case `| PatTyped (p, t) ->` I think.

Reply:
Okay that does seem to fix those tests for me, but it makes a bunch of other tests fail now in a similar way as the change in `genField`. Would we want to try and make it only do this for record types with `MultilineBlockBracketsOnSameColumn`? If so how would be the best way to handle that check in these places?